syntax = "proto3";

option java_multiple_files = true;
option java_package = "org.qsn.api.rpc.grpc.proto.entity.request";

package org.qsn.protobuf;

import "qsn/entity/common/BaseRequest.proto";
import "qsn/entity/common/BInteger.proto";
import "qsn/entity/common/transaction/TransactionDto.proto";

message SendTransactionRequest{
    BaseRequest baseRequest = 1;
    TransactionDto transactionDto = 2;
}

message GetTransactionReceiptRequest{
    BaseRequest baseRequest = 1;
    bytes transactionHash = 2;
}

message GetTransactionRequest{
    BaseRequest baseRequest = 1;
    bytes transactionHash = 2;
}

message CreateRawTransferTxRequest{
    BaseRequest baseRequest = 1;
    bytes from = 2;
    bytes to = 3;
    BInteger amount = 4;
}

message CreateRawDeployContractTxRequest{
    BaseRequest baseRequest = 1;
    bytes contractCode = 2;
    string contractName = 3;
    string contractClassName = 4;
}

message CreateRawValidatorTxRequest {
    BaseRequest baseRequest = 1;
    bool  type = 2;
    bytes validatorAddress = 3;
    bytes validatorPubKey = 4;
    bytes delegationAddress = 5;
    BInteger stake = 6;
    string name = 7;
}

message CreateRawWithdrawTxRequest{
    BaseRequest baseRequest = 1;
    bytes coinBase = 2;
    BInteger value = 3;
}
